types/n3/index.d.ts

```diff
@@ -13,87 +13,104 @@ import * as RDF from "rdf-js";
+export class BlankNode implements RDF.BlankNode {
     static nextId: number;
     termType: "BlankNode";
     value: string;
     constructor(name: string);
+    id: string;
+    toJSON(): {};
+    equals(other: Term): boolean;
+    static subclass(type: any): void;
 }
 
-export class Variable extends Term implements RDF.Variable {
+export class Variable  implements RDF.Variable {
     termType: "Variable";
     value: string;
     constructor(name: string);
+    id: string;
+    toJSON(): {};
+    equals(other: Term): boolean;
+    static subclass(type: any): void;
 }
 
-export class Literal extends Term implements RDF.Literal {
+export class Literal implements RDF.Literal {
     static readonly langStringDatatype: NamedNode;
     termType: "Literal";
     value: string;
+    id: string;
+    toJSON(): {};
+    equals(other: Term): boolean;
+    static subclass(type: any): void;
     language: string;
-    datatype: RDF.NamedNode;
+    datatype: NamedNode;
     datatypeString: string;
     constructor(id: string);
 }
 
-export class DefaultGraph extends Term implements RDF.DefaultGraph {
+export class DefaultGraph implements RDF.DefaultGraph {
     termType: "DefaultGraph";
     value: "";
     constructor();
+    id: string;
+    toJSON(): {};
+    equals(other: Term): boolean;
+    static subclass(type: any): void;
 }
 
+export type Quad_Subject = NamedNode | BlankNode | Variable;
+export type Quad_Predicate = NamedNode | Variable;
+export type Quad_Object = NamedNode | Literal | BlankNode | Variable;
+export type Quad_Graph = DefaultGraph | NamedNode | BlankNode | Variable;
+
 export class Quad implements RDF.Quad {
-    constructor(subject: RDF.Term, predicate: RDF.Term, object: RDF.Term, graph?: RDF.Term);
-    subject: RDF.Term;
-    predicate: RDF.Term;
-    object: RDF.Term;
-    graph: RDF.Term;
-    equals(other: RDF.Quad): boolean;
+    constructor(subject: Quad_Subject, predicate: Quad_Predicate, object: Quad_Object, graph?: Quad_Graph);
+    subject: Quad_Subject;
+    predicate: Quad_Predicate;
+    object: Quad_Object;
+    graph: Quad_Graph;
+    equals(other: Quad): boolean;
     toJSON(): string;
 }
 
 export class Triple extends Quad implements RDF.Triple {}
 
 export namespace DataFactory {
-    function namedNode(value: string): RDF.NamedNode;
-    function blankNode(value?: string): RDF.BlankNode;
-    function literal(value: string | number, languageOrDatatype?: string | RDF.NamedNode): RDF.Literal;
-    function variable(value: string): RDF.Variable;
-    function defaultGraph(): RDF.DefaultGraph;
-    function triple(subject: RDF.Term, predicate: RDF.Term, object: RDF.Term): RDF.Quad;
-    function quad(subject: RDF.Term, predicate: RDF.Term, object: RDF.Term, graph?: RDF.Term): RDF.Quad;
+    function namedNode(value: string): NamedNode;
+    function blankNode(value?: string): BlankNode;
+    function literal(value: string | number, languageOrDatatype?: string | NamedNode): Literal;
```

Comment:
```suggestion
    function literal(value: string | number, languageOrDatatype?: string | RDF.NamedNode): Literal;
```
I think this should be a RDF.NamedNode